### PR TITLE
Tab component

### PR DESCRIPTION
**Pull request by Ellie-Brakoniecki**
- Created a new tab component in order to be able to display jitter plots and time series plots for one thematic group (children and vulnerable people, etc.) on the same page
- Follow up ticket for accessibility https://trello.com/c/E2MYllJG/745-investigate-arrow-key-navigations-for-tab-component

image

### CODE REVIEW

File: src/lib/fragments/Tabs.react.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    const selectedTabContent = this.props.children[this.state.tabSelected]
+    let tabHeaders; 
+    if (this.props.children.length === 1 || xtype.type(this.props.children) !== 'array'){                     
+      tabHeaders = this.renderTabHeader(0, this.props.tabHeadings[0]);
+    }
+    else { 
+      tabHeaders = this.props.tabHeadings.map((tabHeader,index) => this.renderTabHeader(index, tabHeader))
+    }
+    return (
+      <div className="app-example-page js-enabled">
```
Comment (Jordan-Hall) [inline src/lib/fragments/Tabs.react.js:29]: Class name should be updated

File: src/lib/fragments/tab.css [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    text-decoration-thickness: 3px;
+    text-underline-offset: 0.3em;   
+}
+
+.govuk-tabs__panel * {
+margin-top:0;   
+}
+
+.cases .govuk-tabs__list-item .govuk-tabs__tab:link {
+    color: #0b0c0c !important;
```
Comment (Jordan-Hall) [inline src/lib/fragments/tab.css:61]: No way of changing this as it's open source you dont want to be enforcing users to use !important to override

File: src/lib/fragments/Tabs.react.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    )
+  }
+
+
+  renderTabHeader(index, tabHeader) {
+    let tabIsSelected;
+    tabIsSelected = (index === this.state.tabSelected) ? true: false
+      return (   
+          <li className={tabIsSelected ? "tab-selected-underline govuk-tabs__list-item govuk-tabs__list-item--selected" : " tab-underline govuk-tabs__list-item"} role="presentation">
+            <button className={tabIsSelected ? "tab-button-selected govuk-tabs__tab":  "tab-button govuk-tabs__tab"} role = "tab" type="button" aria-controls={tabHeader}   aria-labelledby={tabHeader} aria-selected={tabIsSelected ? 'true':'false'} onClick={() => this.setSelectedTabIndex(index)}> {tabHeader}
```
Comment (Jordan-Hall) [inline src/lib/fragments/Tabs.react.js:45]: Using onClick doesn't support Government GDS own experimental and not accessible. Ideally you need to have arrow key navigations

https://design-system.service.gov.uk/components/tabs/

**Review by Jordan-Hall — CHANGES_REQUESTED** [review]
Left some comments. Sorry i still keep an eye, Happy for it to go in if required but its seems to be missing the keyboard navigation which is key to GDS

File: src/lib/fragments/Tabs.react.js (hunk 3 above)
Comment (kateriley1) [reply src/lib/fragments/Tabs.react.js:45]: Thanks Jordan - I'm going to create another spike ticket to look into key navigations.

File: src/demo/App.js [hunk 4]
```diff
@@ -1,7 +1,8 @@
 /* eslint no-magic-numbers: 0 */
 import React, { useState }  from 'react';
 
-import { Dropdown, CheckboxList, ComponentTemplate, ExpandableMenuItem, Accordion } from '../lib';
+import { Dropdown, CheckboxList, ComponentTemplate, ExpandableMenuItem, Accordion, Tab2 } from '../lib';
```
Comment (tom-clarke-dluhc) [inline src/demo/App.js:4]: Is Tab2 still needed?
```suggestion
import { Dropdown, CheckboxList, ComponentTemplate, ExpandableMenuItem, Accordion } from '../lib';

```

**Review by tom-clarke-dluhc — APPROVED** [review]

**Pull request merged**